### ISSUE TITLE
knife cookbook upload --all does not upload all cookbooks if any one of the cookbook version is frozen.

### ISSUE THREAD

**Issue by Vasu1105** [chef/chef#9157]
## Description

``` knife cookbook upload --all ``` does not upload all the cookbooks if anyone of the cookbook's version is frozen. And as per current behaviour knife cookbook upload --alll shows wrong message at the end of command execution i.e. ``` Uploaded all cookbooks. ```  which confuse the user.

We either need to fix the message or the functionality to fix this.

## Chef Version

```
ChefDK version: 4.6.35
Chef Infra Client version: 15.5.17
Chef InSpec version: 4.18.39
Test Kitchen version: 2.3.4
Foodcritic version: 16.2.0
Cookstyle version: 5.13.7
```
Present in Chef 14 too.
## Replication Case

I have one of the cookbook in repo whose version is frozen.
Then I try to upload all cookbooks from my cookbook repo using --all option
 
## Client Output

```

knife cookbook upload --all
Uploading berks-test [0.1.0]
Uploading one-more-test [0.1.0]
Uploading one-moret-test [0.1.0]
Uploading starter [1.3.0]
Uploading test-cookbook [0.1.0]
Uploading test-one [0.1.0]
Uploading test-two [0.1.0]
ERROR: Version 0.1.0 of cookbook berks-test is frozen. Use --force to override.
WARNING: Not updating version constraints for some cookbooks in the environment as the cookbook is frozen.
Uploaded all cookbooks.

```

**Comment by lamont-granquist** [comment]
that seems to be the correct behavior though.  frozen cookbooks need to never be overwritten unless the user specifies `--force`.

seems like a semantic issue where "uploaded all cookbooks" really means "uploaded all the cookbook that i could".

**Comment by Vasu1105** [comment]
@lamont-granquist it does not upload any of the cookbook but says "Uploaded all cookbooks"

**Comment by Vasu1105** [comment]
Adding more command execution logs
```
$ knife cookbook list
berks-test       0.1.0
one-moret-test   0.1.0
registry_key     0.1.0
starter          1.3.0

$ knife cookbook upload starter
Uploading starter        [1.3.0]
ERROR: Version 1.3.0 of cookbook starter is frozen. Use --force to override.
WARNING: Not updating version constraints for starter in the environment as the cookbook is frozen.
ERROR: Failed to upload 1 cookbook.

$ knife cookbook upload --all
Uploading one-more       [0.1.0]
Uploading one-more-test  [0.1.0]
Uploading one-moret-test [0.1.0]
Uploading starter        [1.3.0]
Uploading test-cookbook  [0.1.0]
Uploading test-one       [0.1.0]
Uploading test-two       [0.1.0]
ERROR: Version 1.3.0 of cookbook starter is frozen. Use --force to override.
WARNING: Not updating version constraints for some cookbooks in the environment as the cookbook is frozen.
Uploaded all cookbooks.

$ knife cookbook list
berks-test       0.1.0
one-more         0.1.0
one-more-test    0.1.0
one-moret-test   0.1.0
registry_key     0.1.0
starter          1.3.0
```

**Comment by lamont-granquist** [comment]
That's still very low priority.  People should really be using Berkshelf or PolicyFiles, and if they're trying to upload everything they can still see the error and override it with `--force` which is probably what they're going to want to do.  Until a user error comes in based on this, I'm not sure what the correct behavior would be.

**Comment by Vasu1105** [comment]
Its fixed as UX in #9073
